python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -985,8 +985,14 @@ def placement_groups_to_resource_demands(
     """
     resource_demand_vector = []
     unconverted = []
+    shapes = []
     for placement_group in pending_placement_groups:
-        shapes = [dict(bundle.unit_resources) for bundle in placement_group.bundles]
+        # Skip **placed** bundle (which has node id associated with it).
+        for bundle in placement_group.bundles:
+            if bundle.node_id:
```

Comment:
Is it an empty string or `None`? If it is `None`, use `is` instead. 

```suggestion
            if bundle.node_id is not None:
```

Reply:
fix in [7a44207](https://github.com/ray-project/ray/pull/48924/commits/7a44207b496f2644f64031aca6c4eefc7223f364), it should be an empty byte string.